linked_list/main.py

```diff
@@ -62,34 +68,116 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        
+        if elements:
+            
+            self.end = Node(elements[-1])
+            node = Node(elements[0], None)
+            last_node = self.end
+            for element in reversed(elements[:-1]):
+                node = Node(element, last_node)
+                last_node = node
+            self.start = node
+    
 
     def __str__(self):
-        pass
-
+        lst = [str(el) for el in self]
+        
+        return "[" + ", ".join(lst) + "]"
+        
     def __len__(self):
-        pass
+        return self.count()
 
     def __iter__(self):
-        pass
+        return (node.elem for node in self.iterate_nodes())
+        
 
     def __getitem__(self, index):
-        pass
+        if index < 0:
+            index = len(self) + index
+            
+        for count, elem in enumerate(self):
+            if count == index:
+                return elem
+
+            
 
     def __add__(self, other):
-        pass
+        
+        new_list = [el for el in self]
+        new_list += [el for el in other]
+
+        return LinkedList(new_list)
 
     def __iadd__(self, other):
-        pass
+        
+        for el in other:
+            self.append(el)
+            
+        return self
 
     def __eq__(self, other):
-        pass
 
+        for el1, el2 in zip(self, other):
+            if el1 != el2:
+                return False
+                
+        return len(self) == len(other)
+                
+    def iterate_nodes(self):
```

Comment:
Lets say you are using this list, wouldn't you expect it to give back the values you put into it when you iterate over it? Thats also how the built in list behaves in python.